bp/src/Bluepill.m

```diff
@@ -485,16 +486,80 @@ - (BOOL)isProcessRunningWithContext:(BPExecutionContext *)context {
+            NSString *screenshotsTestDirectory = [NSString stringWithFormat:@"%@/%@", screenshotsRootDirectory, context.runner.device.UDID];
+            NSURL *screenshotsTestURL = [NSURL fileURLWithPath:screenshotsTestDirectory];
+            
+            [BPUtils printInfo:INFO withString:@"Creating test screenshots directory: %@", screenshotsTestDirectory];
+            
+            NSError *createDirectoryError = nil;
+            [fileManager
+                   createDirectoryAtURL:screenshotsTestURL
+            withIntermediateDirectories:YES
+                             attributes:nil
+                                  error:&createDirectoryError];
+            
+            if (createDirectoryError) {
+                [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"Failed to create directory for screenshots: %@", [createDirectoryError localizedDescription]]];
+            }
+            
+            NSString *dataPath = context.runner.device.dataPath;
+            NSString *failureScreenshotsPath = [NSString stringWithFormat:@"%@/Containers/Data/Application", dataPath];
+            [BPUtils printInfo:INFO withString:@"Grabbing screenshots from: %@", failureScreenshotsPath];
+            
+            NSURL *directoryURL = [NSURL fileURLWithPath:failureScreenshotsPath]; // URL pointing to the directory you want to browse
+            NSArray *keys = [NSArray arrayWithObject:NSURLIsDirectoryKey];
+
+            NSDirectoryEnumerator *enumerator = [fileManager
+                enumeratorAtURL:directoryURL
+                includingPropertiesForKeys:keys
+                options:0
+                errorHandler:^(NSURL *url, NSError *error) {
+                    // Handle the error.
+                    // Return YES if the enumeration should continue after the error.
+                    return YES;
+            }];
 
-        [BPUtils printInfo:INFO withString:@"Writing JUnit report to: %@", outputFile];
-        BPWriter *junitLog = [[BPWriter alloc] initWithDestination:BPWriterDestinationFile andPath:outputFile];
-        [junitLog writeLine:@"%@", [context.parser generateLog:[[JUnitReporter alloc] init]]];
-        [context.parser cleanup];
+            for (NSURL *url in enumerator) {
+                NSError *error;
+                NSNumber *isDirectory = nil;
+                if (! [url getResourceValue:&isDirectory forKey:NSURLIsDirectoryKey error:&error]) {
+                    // handle error
+                    [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"Unable to get screenshot file from: %@, error: %@", url.absoluteString, [error localizedDescription]]];
+                } else if (! [isDirectory boolValue]) {
+                    // No error and it’s not a directory; do something with the file
+                    if ([url.absoluteString containsString:@"AssertionFailedException"] && [url.pathExtension containsString:@"png"]) {
```

Comment:
Is there an advantage to using the testing framework to do snapshots instead of taking them from bluepill bluepill? Or is the latter required?

Also, if we can't just naievely take screenshots from bluepill, would it be possible to:
- determine a screenshot path
- write to that screenshot path from earlgrey

It seems like either XCTest or EarlGrey should allow us to give it a screenshot path